Backend-Type-Annotations.md

```diff
@@ -109,5 +109,23 @@ from typing import (
 
 ```
 
+4. While narrowing down the `Optional[...]` types (eg. Optional[str]) always keep assert at your last option, because if assert is not used properly it can create abmbiguity among other developers. For proper usage of assert, you can refer following points:
```

Comment:
I'm seeing some typos and grammar issues, e.g.

```suggestion
4. While narrowing down `Optional[...]` types like `Optional[str]`, always keep assert as your last option because improper asserts can create ambiguity for other developers. To use asserts properly, please follow these points:
```

ditto elsewhere.